algoliasearch-tests/src/test/java/com/algolia/search/SyncAlgoliaIntegrationTest.java

```diff
@@ -74,7 +74,7 @@ protected Rule generateRule(String objectID) {
     Consequence ruleConsequence =
         new Consequence()
             .setUserData(ImmutableMap.of("a", "b"))
-            .setParams(new ConsequenceParams().setFacets("a=1"));
+            .setParams(new ConsequenceParams<>().setFacets("a=1"));
```

Comment:
Should we add a comment here as well to explain the trick?

Reply:
I would be careful with this code: since you're using the generic class with type argument, you're accessing the raw class (the type-erased one). It's not clear that you're doing that, or even that it works. I would put the type argument for clarity's sake

Reply:
@Ant-hem Yes good point, I'll add it as well.

@nitnelave In that case, the parametric type is useless. It is only used when some specific setters are called and set the internal `T` members. As we don't play with those members, I guess the inferred type is just `Object` and it's OK in our case.

Reply:
The problem is that the type argument is not inferred, it's erased, and you're accessing the raw type (see http://www.javapractices.com/topic/TopicAction.do?Id=224 for instance). It's a backward compatibility hack that this even compiles. It would be just like write `std:: vector v;` in C++

Reply:
After discussing it internally, we agreed that the most pragmatic choice was to let the diamond operation anyway, except for the places where `setAutomaticFacetFilters` and `setAutomaticOptionalFacetFilters` are called.